tests/time_change/test

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/perl
+
+use strict;
+
+use Test;
+BEGIN { plan tests => 84 }
+
+use File::Temp qw/ tempfile /;
+
+my $basedir = $0;
+$basedir =~ s|(.*)/[^/]*|$1|;
+
+# we need to filter by sessionid to avoid possible interaction with NTP daemon
+my $sid = qx( cat /proc/self/sessionid );
+
+###
+# functions
+
+sub key_gen {
+    my @chars = ( "A" .. "Z", "a" .. "z" );
+    my $key   = "testsuite-" . time . "-";
+    $key .= $chars[ rand @chars ] for 1 .. 8;
+    return $key;
+}
+
+sub filter_setup {
+    my ( $syscall, $key ) = @_;
+    system(
+"auditctl -a always,exit -F sessionid=$sid -F arch=b$ENV{MODE} -S $syscall -k $key"
+    );
+}
+
+sub filter_cleanup {
+    system("auditctl -D >& /dev/null");
+}
+
+###
+# setup
+
+# reset audit
+system("auditctl -D >& /dev/null");
+
+# create stdout/stderr sinks
+( my $fh_out, my $stdout ) = tempfile(
+    TEMPLATE => '/tmp/audit-testsuite-out-XXXX',
+    UNLINK   => 1
+);
+( my $fh_err, my $stderr ) = tempfile(
+    TEMPLATE => '/tmp/audit-testsuite-err-XXXX',
+    UNLINK   => 1
+);
+
+###
+# tests
+
+sub test_settime {
+    my ( $mode, $syscall ) = @_;
+
+    my $key = key_gen();
+    my $result;
+
+    # set the filter
+    filter_setup( $syscall, $key );
+
+    # run the test
+    $result = system("$basedir/test_settime $mode 757094400 123456");
+    ok( $result, 0 );
+
+    # make sure the records had a chance to bubble through to the logs
+    system("auditctl -m syncmarker-$key");
+    for ( my $i = 0 ; $i < 10 ; $i++ ) {
+        if ( system("ausearch -m USER | grep -q syncmarker-$key") eq 0 ) {
```

Comment:
Why are using grep and not `ausearch -k`?

Reply:
Because the marker is a USER record and the key is in its text content, not in the key field.

Reply:
D'oh!  Of course :)  Sorry for the noise.